app/web/features/profile/view/leaveReference/formSteps/submit/SubmitReference.tsx

```diff
@@ -92,25 +99,29 @@ export default function SubmitReference({
       },
       {
         onSuccess: () => {
-          history.push(routeToUser(user.username, "references"));
+          if (userQuery.data) {
+            router.push(routeToUser(userQuery.data.username, "references"));
+          } else {
+            router.push(routeToProfile("references"));
```

Comment:
Would this have been stuck on the end of leave references page before? 🤔

Reply:
I think so yes.